src/mlpack/methods/randomized_svd/randomized_svd.hpp

```diff
@@ -120,6 +137,101 @@ class RandomizedSVD
              arma::mat& v,
              const size_t rank);
 
+  /**
+   * Apply Principal Component Analysis to the provided matrix data set
+   * using the randomized SVD.
+   *
+   * @param data Data matrix.
+   * @param u First unitary matrix.
+   * @param v Second unitary matrix.
+   * @param sigma Diagonal matrix of singular values.
+   * @param rank Rank of the approximation.
+   * @param rowMean Centered mean value matrix.
+   */
+  template<typename mat_type>
+  void Apply(mat_type const& data,
```

Comment:
I think we can simplify the class, and only provide two ``Apply`` methods:

```c++
template<typename MatType>
void RandomizedSVD::Apply(const MatType& data,
                          arma::mat& u,
                          arma::vec& s,
                          arma::mat& v,
                          const size_t rank)
{
  // Center the data into a temporary matrix.
  arma::vec rowMean = arma::sum(data, 1) / data.n_cols + eps;
  Apply(data, u, s, v, rank, rowMean);
}


template<typename MatType>
void RandomizedSVD::Apply(const MatType& data,
                          arma::mat& u,
                          arma::vec& s,
                          arma::mat& v,
                          const size_t rank,
                          const MatType& rowMean)
{
  ...
}
```

do you think that works? Also let's put the implementation inside the ``.cpp class``.

Reply:
On a second look, I would say you already did that, but it looks like the merge failed.